### DISCUSSION TITLE
lagging on levels fixed?

### DISCUSSION THREAD

**Discussion by MTB207** [FunkinCrew/Funkin#3874] [Q&A]
this has probably been said countless times and tbf idk where exactly to ask this but has the lagging issue been resolved or is it just on my end that at crucial points in some levels i just get a major lagspike that basically screws up my whole fc?
like ive been trying to fc cocoa erect since it released and its still impossible cus the game just lags at a specific point
why does it happen and can it be fixed? if not, is there an estimate for when this will get fixed? i am sure that the funkin crew is hard at work and is doing their absolute best to resolve any and all bugs existent currently but god damn it man i just wanna get my fc on this one level.
thank you for your time, whoever decided to read this (if you even exist), and my apologies for misusing the discussions/compiling help section if that is the case.

**Comment by KoloInDaCrib** [comment]
Lags often happen every time the function [`resyncVocals` in `PlayState.hx`](https://github.com/FunkinCrew/Funkin/blob/9e13ea0b2c0799d9a7fae7db56eddb05898739d0/source/funkin/play/PlayState.hx#L2126) gets called, which is (among other circumstances like unpausing) every beat. That is if the difference between the current position in one of the given sound files and the conductor's songPosition variable is bigger than 5 milliseconds (offsets are accounted for in the calculation). For whatever reason in some audio files there are additional offsets that are not accounted for anywhere (I'm guessing due to some bitrate thing or whatever idk), which makes the difference 10ms. (11ms in case of Fresh (Pico Mix)).

There is a [Pull Request with the "fix"](https://github.com/FunkinCrew/Funkin/pull/3735) (I literally just bumped the treshold to 15ms lol) ongoing but it's yet to be reviewed by the funkin crew though it may even be fixed internally, not sure ¯\_(ツ)_/¯

**Comment by MAZ12211** [comment]
The lag happens on my end too dw
It's annoying, hope they fix it or have fixed it

**Comment by Beelijah** [comment]
this is the only reason i play on v0.5.0 instead of the latest version (v0.5.3)

**Reply by MAZ12211** [reply]
😅

**Reply by KoloInDaCrib** [reply]
I wouldn't recommend playing on 0.5.0, as it suffers from an even bigger issue - incorrectly applied offsets, causing the game to be literally unplayable
